From 5fc95412e13a73df6a701baad1ac9b5d2a5beeb8 Mon Sep 17 00:00:00 2001
From: Kenny Kessler <kenny.kessler@motorola.com>
Date: Wed, 16 Mar 2016 11:28:30 -0500
Subject: [PATCH 799/959] IKSWM-25798: power: qpnp-smbcharger-mmi: Lower HVDCP
 ICL

This baselines should not support greater then a 15W HVDCP
Remove configuring of the Input Current Limit to 2600 mA
and leave it at 1600 mA

Change-Id: I1b5ead5cb362e929645a00f71c27863913b2727b
Signed-off-by: Kenny Kessler <kenny.kessler@motorola.com>
Reviewed-on: http://gerrit.mot.com/833858
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index f1a709a1968..d0f9058ce7d 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -3773,15 +3773,8 @@ static void smbchg_external_power_changed(struct power_supply *psy)
 	else
 		current_limit = prop.intval / 1000;
 
-	if (smbchg_hvdcp_det_check(chip)) {
-		if ((chip->stepchg_state == STEP_TAPER) ||
-		    (chip->stepchg_state == STEP_FULL) ||
-		    (chip->usb_target_current_ma == HVDCP_ICL_TAPER) ||
-		    (chip->demo_mode))
-			current_limit = HVDCP_ICL_TAPER;
-		else
-			current_limit = HVDCP_ICL_MAX;
-	}
+	if (smbchg_hvdcp_det_check(chip))
+		current_limit = HVDCP_ICL_TAPER;
 
 	pr_smb(PR_MISC, "current_limit = %d\n", current_limit);
 	mutex_lock(&chip->current_change_lock);
-- 
2.11.0

